### ISSUE TITLE
Test low mAP 

### ISSUE THREAD

**Issue by matbord** [ultralytics/yolov5#2858]
## ❔Question
Hi, I'm training my own dataset for object detection from drone's videos. When I test my video using my trained model I'm getting very low mAP: mAP@.5 0.213    mAP@.5:.95  0.0397 but the detection doesn't result so bad. 
So I took the code to calculate the mAP from this repo https://github.com/Cartucho/mAP and I obtained a mAP of 0.805.
I would like to know if Yolov5's mAP is calculated in a different way compared to the common one.

**Comment by glenn-jocher** [comment]
@matbord mAP is computed in metrics.py and verified to about 1% similarity to pycocotools mAP on the COCO val and test datasets.
https://github.com/ultralytics/yolov5/blob/1df8c6c963d31ce84895101a70e45e0afdcb0bc2/utils/metrics.py#L18-L107

I'm not aware of any mAP issues other than the 1% difference with pycocotools, but if you spot any specific issues please let us know!

**Comment by glenn-jocher** [comment]
@SkalskiP this issue might interest you. I also glanced at https://github.com/rafaelpadilla/Object-Detection-Metrics quickly and see that they produced a revised set of tools about 5 months ago at https://github.com/rafaelpadilla/review_object_detection_metrics

**Comment by matbord** [comment]
Thank you for your answer. Probably my problem is that I am comparing the mAP@.5 and mAP@.5:.95  with the normal mAP. If I want to change the IoU Threshold I'm obtaining different results because they are calculated in a different way. Do you agree?

**Comment by matbord** [comment]
@glenn-jocher Thank you for your answer. Probably my problem is that I am comparing the mAP@.5 and mAP@.5:.95 with the normal mAP. If I want to change the IoU Threshold I'm obtaining different results because they are calculated in a different way. Do you agree?

**Comment by SkalskiP** [comment]
@glenn-jocher thanks, yes you are correct that issue is interesting to me ;) I know about that repo you linked, I'm still thinking bigger than the only CV so I'm not overly stressed about it.

**Comment by glenn-jocher** [comment]
@matbord we compute mAP at 10 intervals for IoU thresholds between 0.5 and 0.95. The ten mAPs are averaged for mAP@0.5:0.95, and the first interval is used for mAP@0.5. I believe this is very much standard practice and all mAP computation tools will do some version of the same steps.

**Comment by matbord** [comment]
@glenn-jocher ok, thankyou I understand the explanation about mAP@0.5:0.95 and mAP@0.5. But if I want to calculate only a mAP (not mAP@0.5:0.95 or mAP@0.5) using an IoU with a threshold of 0.2 which function do I have to use?

**Comment by glenn-jocher** [comment]
@matbord I don't understand, you want mAP@0.2? I've not heard this request before, this is not a commonly used metric.

You can modify the IoU vector used for mAP here if you'd like. The first threshold is the one assigned to the mAP@0.5 output in test.py:
https://github.com/ultralytics/yolov5/blob/c949fc86d1914bdbf0a61d193855c1b4e1536da5/test.py#L77

**Comment by matbord** [comment]
@glenn-jocher I want to compare Yolov5 results with RetinaNet results. RetinaNet offers a mAP with IOU Threshold t= (my value). Yolo offers only mAP@0.5:0.95 or mAP@0.5 which are calculated in a different way compared to mAP with IOU Threshold of RetinaNet. I would like to compare the 2 results in order to decide which algorithm is better but I cannot do it if the 2 metrics are different.
This is why I used  https://github.com/Cartucho/mAP mAP calculus and adapted YoloV5 annotations to calculate it. 

Anyway, I just tried and changing the line you mentioned I obtained exactly the same value using  https://github.com/Cartucho/mAP . So it is what I was searching, thankyou! :)
